openapi3.yaml

```diff
@@ -1,57 +1,143 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        404:
+          description: NOT FOUND
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+
+  /addProduct:
+    post:
+      operationId: addProduct
       tags:
-        - resourceName
-      summary: gets the resource
+        - addProduct
+      summary: creates a new record of type product
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'          
+      responses:
+        201:
+          description: created, send back the product that added
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/product'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+
+  /updateProduct/{id}:
+    patch:
+      operationId: updateProduct
+      tags:
+        - updateProduct
+      summary: update the chosen feilds of a product
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'      
       responses:
         200:
-          description: OK
+          description: updated, send back the product that updated
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/resource'
-    post:
-      operationId: createResource
+                $ref: '#/components/schemas/product'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+
+  /deleteProduct/{id}:
+    delete:
+      operationId: deleteProduct
       tags:
-        - resourceName
-      summary: creates a new record of type resource
+        - deleteProduct
+      summary: delete the a product by its id   
       responses:
-        201:
-          description: created
+        200:
+          description: deleted, send back the product that deleted
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/resource'
+                $ref: '#/components/schemas/product'
```

Comment:
When `deleting` a product it is returned in the response? Why?

Reply:
just to let the user make sure everything worked as expected.